src/jit/lsraarmarch.cpp

```diff
@@ -679,7 +679,7 @@ int LinearScan::BuildBlockStore(GenTreeBlk* blkNode)
         }
     }
 
-    if ((size != 0) && (blkSizeRegMask != RBM_NONE))
+    if (!blkNode->OperIs(GT_STORE_DYN_BLK) && (blkSizeRegMask != RBM_NONE))
```

Comment:
If would be nicer to have `if (blkNode->OperIs(GT_STORE_DYN_BLK)) { … } else { … }` than the current 2 `if`s but I can do that in my block op PR (if I haven't done it already).

Reply:
It wouldn't be correct to combine them without duplicating code, because the `RefPosition`s need to retain the correct order.